src/runtime/scheduler/scheduler.rs

```diff
@@ -195,11 +195,11 @@ mod tests {
 
         // All futures are inserted in the scheduler with notification flag set.
         // By polling once, our future should complete.
-        if let Some(_) = scheduler.poll_group_once(group_id, None).pop() {
-            Ok(())
-        } else {
+        if scheduler.poll_group_once(group_id, None).pop().is_none() {
```

Comment:
I think this could be further turned into an ok_or or an ok_or_else: https://doc.rust-lang.org/std/option/enum.Option.html#method.ok_or_else

Reply:
That would be great, however `bail!()` does not return any value, it just returns early from the fn. `ok_or` or `ok_or_else` require a return value from the expression. We could change from `anyhow::bail!()` to `anyhow::anyhow()` to construct the error, but then we would end up changing the semantics of this code. Do you prefer any particular style in this case?